src/Bundler.js

```diff
@@ -323,22 +323,33 @@ class Bundler extends EventEmitter {
     }
 
     // Process asset dependencies
-    await Promise.all(
+    const applyFunctions = await Promise.all(
       dependencies.map(async dep => {
+        let applyFunction;
         let assetDep = await this.resolveDep(asset, dep);
         if (dep.includedInParent) {
           // This dependency is already included in the parent's generated output,
           // so no need to load it. We map the name back to the parent asset so
           // that changing it triggers a recompile of the parent.
-          this.loadedAssets.set(dep.name, asset);
+          applyFunction = () => this.loadedAssets.set(dep.name, asset);
         } else {
-          asset.dependencies.set(dep.name, dep);
-          asset.depAssets.set(dep.name, assetDep);
+          applyFunction = () => {
+            asset.dependencies.set(dep.name, dep);
+            asset.depAssets.set(dep.name, assetDep);
+          };
           await this.loadAsset(assetDep);
         }
+        return applyFunction;
       })
     );
```

Comment:
Is `applyFunctions` only job to be run later. Then would it be leaner to chain `.forEach(fn => fn())` do you think?

Reply:
That would work as well. I felt the current was more readable.

Reply:
All good. Confirmed test is stable on my end now.